### PR TITLE
Update entries for Beej's Guides

### PR DESCRIPTION
**Pull request by jayaddison**
## What does this PR do?

For Beej's Guide to Networking ([canonical link](https://www.beej.us/guide/bgnet/)):

- Adds 1x reference within the books-by-subject (section: Networking)
- Updates 1x reference with file format availability in books-by-programming-language (section: C)
- Adds 4x references to translations of the book per-natural-language (sections: Dansk, Magyar, Române, Srpski)

From code review feedback:

- Updates additional and existing references to author 'B. Hall' to use full name 'Brian "Beej Jorgensen" Hall'

### Why is this valuable (or not)?
"Beej's Guide to Networking" is a valuable guide to network programming in C.

### How do we know it's really free?
The work is currently "Attribution-NonCommercial-NoDerivs 3.0" licensed, as indicated in the publication's source: https://github.com/beejjorgensen/bgnet/blob/cce225ad6856b749bf92a81ed26208785ed5adb9/src/bgnet.md#copyright-distribution-and-legal-legal

One exception to this is that the author requires:

"One specific exception to the "No Derivative Works" portion of the license is as follows: this guide may be freely translated into any language, provided the translation is accurate, and the guide is reprinted in its entirety. The same license restrictions apply to the translation as to the original guide. The translation may also include the name and contact information for the translator."

All of the translations linked to here are listed on the [author's own website for the book](https://www.beej.us/guide/bgnet/).

### For book lists, is it a book? For course lists, is it a course? etc.
This is a book; it's canonically available in English on the web in HTML and PDF formats, and also has published paper formats, and a number of online translations available.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- [x] Check the status of GitHub Actions and resolve any reported warnings!

cc @beejjorgensen (author)

### CODE REVIEW

File: books/free-programming-books-hu.md [hunk 1]
```diff
@@ -37,6 +37,11 @@
 * [Arduino programozási kézikönyv](http://avr.tavir.hu) - Brian W. Evans írása alapján fordította, kiegészítette és frissítette Cseh Róbert (PDF - regisztráció szükséges)
 
 
+### C
```
Comment (davorpa) [inline books/free-programming-books-hu.md:40]: Needs entry at Index TOC section

File: books/free-programming-books-ro.md [hunk 2]
```diff
@@ -13,6 +13,11 @@
 * [Ajax](http://etutoriale.ro/articles/1483/1/Tutorial-Ajax/)
 
 
+### C
+
+* [Ghidul Beej pentru Programarea in Retea - Folosind socket de internet](https://web.archive.org/web/20180710112954/http://weknowyourdreams.com/beej.html)
```
Comment (davorpa) [inline books/free-programming-books-ro.md:18]: Needs entry at Index TOC section and add the archived notes

File: books/free-programming-books-sr.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+### Index
+
+* [C](#c)
+
+
+### C
+
+* [Beej-ov vodič za mrežno programiranje - Korištenje Internet soket-a](https://web.archive.org/web/20181008134854/http://users.teol.net:80/~mvlado/sockets/) - B. Hall (HTML)
```
Comment (davorpa) [inline books/free-programming-books-sr.md:8]: New files should have its entry at README.md

Locale is OK. Serbian: https://www.localeplanet.com/icu/sr/index.html

**Review by davorpa — CHANGES_REQUESTED** [review]
Moreover, when we prefer the full name of author: `Brian "Beej Jorgensen" Hall`

we should provide the name of the traducer in translation entries too when would be posible

File: books/free-programming-books-ru.md [hunk 4]
```diff
@@ -134,7 +134,7 @@
 ### C
 
 * [Заметки о языке программирования Си/Си++](https://yurichev.com/writings/C-notes-ru.pdf) - Денис Юричев (PDF)
-* [Краткое руководство Beej к GDB](https://paintingvalley.com/ru-bggdb) - (HTML)
+* [Краткое руководство Beej к GDB](https://paintingvalley.com/ru-bggdb) - Brian "Beej Jorgensen" Hall (HTML)
```
Comment (jayaddison) [inline books/free-programming-books-ru.md:137]: (note: it wasn't clear to me who the translation was provided by for this existing entry; I considered attempting to add a credit along with the name change)

**Comment by jayaddison** [comment]
Please note: there is an additional translation in Farsi that is not yet added here.  I'm unable to read that language at the moment and wasn't able to determine what the title of the translated book was.

File: books/free-programming-books-subjects.md [hunk 5]
```diff
@@ -566,6 +566,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 ### Networking
 
 * [An Introduction to Computer Networks](http://intronetworks.cs.luc.edu) (HTML, PDF, Kindle)
+* [Beej's Guide to Network Programming - Using Internet Sockets](https://www.beej.us/guide/bgnet/) - Brian "Beej Jorgensen" Hall (HTML, PDF)
```
Comment (eshellman) [inline books/free-programming-books-subjects.md:569]: we don't duplicate links in multiple files. I think this fits better in Network programming than in "C".  We recently started to add cross-links where this makes sense #5535

**Review by eshellman — CHANGES_REQUESTED** [review]
Wow this is some great work!

**Review by davorpa — CHANGES_REQUESTED** [review]

File: books/free-programming-books-subjects.md (hunk 5 above)
Comment (jayaddison) [reply books/free-programming-books-subjects.md]: Ok, that makes sense - I've removed the C-language entry.  I haven't added a cross-link for this but I'll have a read of the proposals in there.

**Review by eshellman — APPROVED** [review]

**Comment by jayaddison** [comment]
Thanks @eshellman!

**Review by davorpa — APPROVED** [review]
LGTM

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**